Migrate to 1Password (#58)
helpfulengineering/devops: guidance/credential-sharing.md

# Helpful Engineering Shared Credentials

Whenever possible, no account should be shared with other users.  However, in some instances (e.g. social media accounts, some email accounts, *et cetera*) it will be required to share a single set of credentials with multiple users by using a tool called **[1Password](https://1password.com)**.

> The DevOps and Infosec teams maintain a master account where all credentials are stored and shared with the approved users. Passwords are being rotated regularly for security reasons, and the new ones will automatically synchronize in the other users' 1Password vaults. 

Author: Currently it looks like DevOps and Infosec are nearly like one big team; I would dare<sup>1</sup> to say we're nearly DevSecOps (in no particular order).

***
<sub>1. @dmangot has very concrete opinions about `/.*Dev.*Ops.*/` extensions, and there are probably better names to describe this mix of teams. 😄</sub>

Reviewer: I think the names of the teams are fine, but I don't think that deleting the old text in favor of the new one sufficiently conveys the importance of not sharing the passwords by other means.  You're leaving as a mental exercise for the reader to come to that conclusion which seems obvious to you with your background, but may not be obvious to them.

Author: I think I haven't deleted the text in red, but it [has just moved](https://github.com/helpfulengineering/devops/pull/58/files#diff-30163411486162d996903735364b2328R7). The `diff` comes from the mentioned line number, but it's showing all the changes for that line (including relocations).

Should we mark this conversation as resolved?

Author: May we mark this conversation as solved?

:warning: Credentials should **never** be shared outside the password manager: please **do not** send passwords in plaintext by any other means.

## :book: Getting started

> Just send your email address to the _Infosec_ team ([#skill-infosec](https://app.slack.com/client/TUTSYURT3/CV4TYGC1Z)) and you'll be added to 1Password with access to the shared secrets you're allowed to use. If you already have a 1Password account, use its email address when requesting access.

Author: The old document stated that we would be collecting the users' email addresses through the public [#skill-infosec](https://app.slack.com/client/TUTSYURT3/CV4TYGC1Z) channel, but this seems like a bad security measure. Wouldn't be better to ask for direct messages?

Reviewer: I don't think security through obscurity is a principle upon which we should rely.

Author: Which measure do you think is "security by obscurity"? I think that posting people's email addresses in a public channel is not a good practice, and, as we're hiding emails from the Slack profiles, we should also discourage publishing the email addresses on an open channel.

Reviewer: We're not hiding the emails for security reasons, but for privacy reasons.  I think the instructions need to be more clear about the fact that it's a DM and how that could be done.

Author: > We're not hiding the emails for security reasons, but for privacy reasons.

👍 Good point. It definitely was a lax word/concept choice<sup><sup>1</sup></sup>

>  I think the instructions need to be more clear about the fact that it's a DM and how that could be done.

I haven't changed the suggested method yet, so we could choose changing it to a direct message or even a Google/Airtable form. I don't know which method is going to offer a better user experience, but Slack-based communications allows us ensure that the provided emails effectively come from who are supposed to.

***
<sub>1. ...though in some (edge) cases, exposed email addresses might also (indirectly) affect the security: is so easy to lookup leaked passwords with an email address and perform targeted attacks.</sub>

Just send your email address to the _Infosec_ team ([#skill-infosec](https://app.slack.com/client/TUTSYURT3/CV4TYGC1Z)) and you'll be added to 1Password with access to the shared secrets you're allowed to use. If you already have a 1Password account, use its email address when requesting access.

[Lastpass sharing documentation](https://helpdesk.lastpass.com/sharing-4-0/)
If you aren't part of the [@admin-group](https://helpfulengineering.slack.com/admin/user_groups) in Slack, approval for access will be required.

## Changing passwords
When changing a password, you should [generate the new one with LastPass](https://helpdesk.lastpass.com/generating-a-password/) (or equivalent) using the following settings:
Now, you can proceed to set up your browser and devices following this guide: https://support.1password.com/explore/team-member/

> * **All characters**:

Author: Deleted the password generation settings because the new password manager has pretty sensible defaults

# Helpful Engineering Shared Credentials

Whenever possible, no account should be shared with other users. However, in some instances (e.g. social media accounts, some email accounts, *et cetera*) it will be required to share a single set of credentials with multiple users by using a tool called [1Password](https://1password.com).

> The DevOps and Infosec teams maintain a master account to store, manage and share the credentials with the approved users. Thereby, passwords will be automatically synchronized with the other users' 1Password vaults when they are periodically rotated for security reasons.

Author: ```suggestion
The DevOps and Infosec teams maintain a master account to store, manage and share the credentials with the approved users. Thereby, passwords will be automatically synchronized with the other users' 1Password vaults when they change.
```

Maybe we should make the sentence a bit more general.
_Note: the suggested change `diff` is intended_
